mpegts-segmenter/src/segmenter.rs

```diff
@@ -143,6 +143,9 @@ impl<S: SegmentStorage> Segmenter<S> {
                                                 }
                                             }
                                         }
+                                        Some(analyzer::Stream::ADTSAudio { .. }) => {
+                                            is_keyframe = elapsed_seconds < -1.0 || elapsed_seconds >= self.config.min_segment_duration.as_secs_f64().max(3.0);
```

Comment:
This could potentially impact audio+video streams too, right?

Could we make this logic only apply for streams that don't have video?

Reply:
I think for audio+video streams, if the mpegts packet is an audio packet, the following condition is false, so `should_start_new_segment` is not determined by the audio in this case.

```rust
      if (self.analyzer.is_video(p.packet_id) || !self.analyzer.has_video())
         && (elapsed_seconds < -1.0 || elapsed_seconds >= self.config.min_segment_duration.as_secs_f64())
```

Reply:
Ah, I think you're right. Maybe a comment is warranted this:

```suggestion
                                            // note this is only reachable if this is an audio-only stream
                                            is_keyframe = elapsed_seconds < -1.0 || elapsed_seconds >= self.config.min_segment_duration.as_secs_f64().max(3.0);
```

Reply:
This was my question too earlier in standup @lzuosym Chris phrased it better.